### PR TITLE
CMR-8551 "RelatedUrls > MimeType" validation returns a Warning and no…

### PR DESCRIPTION
**Pull request by eriknelson11**
…t an Error

### CODE REVIEW

File: ingest-app/src/cmr/ingest/validation/validation.clj
Comment (jceaser) [inline ingest-app/src/cmr/ingest/validation/validation.clj:104]: I don't know about this, I would think that this function would always return a validator and the issue is if it should be called or not, but I would defer to @sxu123 as she has done the last refactoring of this file. I am leaning in the direction that this is wrong.

File: ingest-app/src/cmr/ingest/validation/validation.clj
Comment (jceaser) [inline ingest-app/src/cmr/ingest/validation/validation.clj:159]: if the original function accepts two parameter signatures, one could assume false and then you would only have to change the code that is different.

File: ingest-app/src/cmr/ingest/validation/validation.clj
Comment (jceaser) [inline ingest-app/src/cmr/ingest/validation/validation.clj:209]: should we not instead use the conditional to conditionally add these functions here at this level? The function name is "optional" it is already deciding what should be shown for a condition.

File: ingest-app/src/cmr/ingest/validation/validation.clj
Comment (jceaser) [inline ingest-app/src/cmr/ingest/validation/validation.clj:105]: double negative? so I send false to get a true? Why not rename to something like report-as-error and when it is true you do the thing? There is also an if-not function, but I think you can make it read better.

File: ingest-app/src/cmr/ingest/validation/validation.clj
Comment (eriknelson11) [reply ingest-app/src/cmr/ingest/validation/validation.clj:104]: to my understanding the first part of `related-url-validator` should be returned as an error but the second part of the function should be returned as a warning, so would a better way be to create a separate function for the warning portion?

File: ingest-app/src/cmr/ingest/validation/validation.clj
Comment (eriknelson11) [reply ingest-app/src/cmr/ingest/validation/validation.clj:105]: `(not warn?)` was used in other parts of the file, I can see how in my case its not as readable though I can change that

Comment (jceaser) [reply ingest-app/src/cmr/ingest/validation/validation.clj:105]: you can get a second opinion, I will defer if I'm the only one

File: ingest-app/src/cmr/ingest/validation/validation.clj
Comment (jceaser) [reply ingest-app/src/cmr/ingest/validation/validation.clj:104]: I think so, but @sxu123 had the most opinions here last time, reach out to her and ask for advice. She reformatted my code and I think it is more readable now.

**Review by sxu123 — CHANGES_REQUESTED** [review]
Sent messages in slack to the developer.

File: ingest-app/src/cmr/ingest/validation/validation.clj
Comment (sxu123) [reply ingest-app/src/cmr/ingest/validation/validation.clj:105]: The logic is right. Although I don't like using the flag because it's a little confusing. I messaged Erik about it.

File: ingest-app/src/cmr/ingest/validation/validation.clj
Comment (sxu123) [reply ingest-app/src/cmr/ingest/validation/validation.clj:104]: Using another function instead of using flag would be more readable.

File: ingest-app/src/cmr/ingest/validation/validation.clj
Comment (sxu123) [reply ingest-app/src/cmr/ingest/validation/validation.clj:159]: Why would you remove           (useconstraints-onlineresource-validators kms-index)
           (collectioncitations-onlineresource-validators kms-index)
           (publicationreferences-onlineresource-validators kms-index))))?
These are not RelatedUrl/MimeType validations. The mime types belong to other fields.

File: ingest-app/src/cmr/ingest/validation/validation.clj
Comment (sxu123) [reply ingest-app/src/cmr/ingest/validation/validation.clj:209]: 1. (useconstraints-onlineresource-validators kms-index)
           (collectioncitations-onlineresource-validators kms-index)
           (publicationreferences-onlineresource-validators kms-index) might not belong here since these MimeTypes don't belong to RelatedUrls.

2. DataCenter are validated twice, on different things. After merging, one would be lost. So They need to be combined in the end.

Comment (sxu123) [reply ingest-app/src/cmr/ingest/validation/validation.clj:209]: A test needs to be added to introduce two errors. One is the DataCenter RelatedUrl/MimeType error, the other is the error from :DataCenters (match-kms-keywords-validation
                           kms-index :providers msg/data-center-not-matches-kms-keywords)})))
Make sure both are returned as warnings, none is lost.

**Review by sxu123 — APPROVED** [review]

**Review by eereiter — APPROVED** [review]

**Pull request merged**
